contracts/mesh-provider/src/state.rs

```diff
@@ -51,6 +61,43 @@ impl Stake {
         val.shares_to_tokens(self.shares)
     }
 
+    /// Calculate rewards
+    pub fn calc_pending_rewards(
+        &mut self,
+        new_rptpb: Uint128,
+        staked: Uint128,
+        height: u64
+    ) -> Result<(), ContractError> {
+        if staked.is_zero() || self.rewards.last_height == 0 {
+            self.rewards.last_rptpb = new_rptpb;
+            self.rewards.last_height = height;
+            return Ok(());
+        }
+
+        let rptpb_to_pay = new_rptpb - self.rewards.last_rptpb;
```

Comment:
What does rptpb mean?
`// Total staked funds, cannot stake more than available funds` ??

This is how much staker S had staked on validator V at `last_height`?

Reply:
sorry, copy paste typo.